docs/docs/flink-writes.md

```diff
@@ -371,3 +371,28 @@ and [deleting orphan files](maintenance.md#delete-orphan-files) could possibly c
 the state of the Flink job. To avoid that, make sure to keep the last snapshot created by the Flink
 job (which can be identified by the `flink.job-id` property in the summary), and only delete
 orphan files that are old enough.
+
+# Flink Writes (SinkV2 based implementation)
+
+The [SinkV2 interface](https://cwiki.apache.org/confluence/display/FLINK/FLIP-191%3A+Extend+unified+Sink+interface+to+support+small+file+compaction)
+was introduced in Flink 1.15.
+The previous [SinkV1 interface](https://cwiki.apache.org/confluence/display/FLINK/FLIP-143%3A+Unified+Sink+API)
+had some limitations - for example it created a lot of small files when writing to it. This problem is called
+the `small-file-compaction` problem in
+the [FLIP-191 document](https://cwiki.apache.org/confluence/display/FLINK/FLIP-191%3A+Extend+unified+Sink+interface+to+support+small+file+compaction).
```

Comment:
Ok, I see your point, but I think we need to change it even more, I'm currently thinking about how to do this. The problem with the proposed changes in the "Introduction" section are:
1. Flink had 3 approaches to defining sink: `SinkFunction`, `SinkV1` (FLIP-143) and `SinkV2` (FLIP-191). In `FlinkSink` we used the first one approach, not the `SinkV1` but ...
2. ... It is not so important because the main logic of sinking logic is done in operators before the sink. For the sink is used DiscardingSink which BTW can be replaced by SinkV2 version to avoid problems with the removal of the previous interface (I've just did it)

IMO we should give some introduction explaining why we have 2 implementations and it is problematic without exposing future plans. Do you see some other idea?

Reply:
I've rewritten the introduction part. Please correct me if I've twisted something or if you see it differently.